tests/load/transformations/test_basic_transformations.py

```diff
@@ -0,0 +1,242 @@
+    @dlt.transformation(
+        write_disposition="replace",
+        table_name="copied_purchases",
+        transformation_type="sql",
+    )
+    def copied_purchases_updated(dataset: SupportsReadableDataset[Any]) -> Any:
+        return dataset["purchases"].limit(3)
+
+    # transform into same dataset
+    dest_p.run(copied_purchases_updated(fruit_p.dataset()))
+    assert row_counts(dest_p.dataset(), ["copied_purchases"]) == {
+        "copied_purchases": 3,
+    }
+
+
+# NOTE: move to duckdb only transformation tests
+@pytest.mark.essential
+@pytest.mark.parametrize(
+    "destination_config",
+    transformation_configs(),
+    ids=lambda x: x.name,
+)
+def test_append_sql_transformations(destination_config: DestinationTestConfiguration) -> None:
+    # get pipelines and populate fruit pipeline
+    fruit_p, dest_p = setup_transformation_pipelines(destination_config, "sql")
+    load_fruit_dataset(fruit_p)
+
+    @dlt.transformation(write_disposition="append", transformation_type="sql")
+    def copied_purchases(dataset: SupportsReadableDataset[Any]) -> Any:
+        return dataset["purchases"].limit(5)
+
+    # transform into same dataset
+    dest_p.run(copied_purchases(fruit_p.dataset()))
+
+    assert row_counts(dest_p.dataset(), ["copied_purchases"]) == {
+        "copied_purchases": 5,
+    }
+
+    @dlt.transformation(
+        write_disposition="append", table_name="copied_purchases", transformation_type="sql"
+    )
+    def copied_table_updated(dataset: SupportsReadableDataset[Any]) -> Any:
+        return dataset["purchases"].limit(7)
+
+    # transform into same dataset
+    dest_p.run(copied_table_updated(fruit_p.dataset()))
+
+    assert row_counts(dest_p.dataset(), ["copied_purchases"]) == {
+        "copied_purchases": 12,
+    }
+
+
+@pytest.mark.skip(
+    reason=(
+        "TODO: lineage got better, so I need a better way to figure out an unknown column type for"
+        " this test. @zilto says all types are always known now, so maybe it is not possible."
+    )
+)
+# NOTE: move to duckdb only transformation tests
+@pytest.mark.essential
+@pytest.mark.parametrize(
+    "destination_config",
+    transformation_configs(),
+    ids=lambda x: x.name,
+)
+def test_sql_transformation_with_unknown_column_types(
+    destination_config: DestinationTestConfiguration,
+) -> None:
+    # get pipelines and populate fruit pipeline
+    fruit_p, dest_p = setup_transformation_pipelines(destination_config, "sql")
+    load_fruit_dataset(fruit_p)
+
+    @dlt.transformation(transformation_type="sql")
+    def mutated_purchases(dataset: SupportsReadableDataset[Any]) -> Any:
+        return dataset["purchases"].mutate(new_col=5).limit(5)
+
+    # problem should already be detected at extraction time
+    with pytest.raises(PipelineStepFailed):
+        dest_p.extract(mutated_purchases(fruit_p.dataset()))
+
+    @dlt.transformation(transformation_type="sql", columns={"new_col": {"data_type": "double"}})
```

Comment:
new_col should be inferred. why do you spevcifu it?

Reply:
yeah, this test is disabled anyway because it was supposed to test what happens if the column type can't be inferred, but I have not found a query yet where sqlglot fails (which is good!) but I can't run the test..